mlflow/pyfunc/__init__.py

```diff
@@ -1226,22 +1224,19 @@ def spark_udf(
 
     model_metadata = Model.load(os.path.join(local_model_path, MLMODEL_FILE_NAME))
 
-    model_output_schema = model_metadata.get_output_schema()
-
     if result_type is None:
-        if model_output_schema is None:
+        if model_output_schema := model_metadata.get_output_schema():
```

Comment:
this change doesn't work.
because the condition is `if model_output_schema is None`